src/aes.rs

```diff
@@ -0,0 +1,227 @@
+        let mut cipher = ctr::Ctr128BE::<aes::Aes128>::new(&self.key.bytes.into(), &self.iv.into());
+        cipher.apply_keystream_b2b(plaintext, buffer).unwrap();
+        Ok(())
+    }
+
+    fn decrypt(&self, ciphertext: &[u8], buffer: &mut [u8]) -> Result<(), signature::Error> {
+        let mut cipher = ctr::Ctr128BE::<aes::Aes128>::new(&self.key.bytes.into(), &self.iv.into());
+        cipher.apply_keystream_b2b(ciphertext, buffer).unwrap();
+        Ok(())
+    }
+}
+
+///
+/// Aes256 in CTR mode
+///
+pub struct Aes256Ctr {
+    pub iv: [u8; 16],
+    pub key: Key<32>,
+}
+
+impl Cipher for Aes256Ctr {
+    fn encrypt(&self, plaintext: &[u8], buffer: &mut [u8]) -> Result<(), signature::Error> {
+        let mut cipher = ctr::Ctr128BE::<aes::Aes256>::new(&self.key.bytes.into(), &self.iv.into());
+        cipher.apply_keystream_b2b(plaintext, buffer).unwrap();
+        Ok(())
+    }
+
+    fn decrypt(&self, ciphertext: &[u8], buffer: &mut [u8]) -> Result<(), signature::Error> {
+        let mut cipher = ctr::Ctr128BE::<aes::Aes256>::new(&self.key.bytes.into(), &self.iv.into());
+        cipher.apply_keystream_b2b(ciphertext, buffer).unwrap();
+        Ok(())
+    }
+}
+
+///
+/// Aes128 in CBC mode with PKCS#7 padding
+///
+pub struct Aes128CbcPkcs7 {
+    pub iv: [u8; 16],
+    pub key: Key<16>,
+}
+
+impl Cipher for Aes128CbcPkcs7 {
+    fn encrypt(&self, plaintext: &[u8], buffer: &mut [u8]) -> Result<(), signature::Error> {
+        let cipher = cbc::Encryptor::<aes::Aes128>::new(&self.key.bytes.into(), &self.iv.into());
+        cipher
+            .encrypt_padded_b2b_mut::<Pkcs7>(plaintext, buffer)
+            .unwrap();
+        Ok(())
+    }
+
+    fn decrypt(&self, ciphertext: &[u8], buffer: &mut [u8]) -> Result<(), signature::Error> {
+        let cipher = cbc::Decryptor::<aes::Aes128>::new(&self.key.bytes.into(), &self.iv.into());
+        cipher
+            .decrypt_padded_b2b_mut::<Pkcs7>(ciphertext, buffer)
+            .unwrap();
+        Ok(())
+    }
+}
+
+///
+/// Aes256 in CBC mode using PKCS#7 padding
+///
+pub struct Aes256CbcPkcs7 {
+    pub iv: [u8; 16],
+    pub key: Key<32>,
+}
+
+impl Cipher for Aes256CbcPkcs7 {
+    fn encrypt(&self, plaintext: &[u8], buffer: &mut [u8]) -> Result<(), signature::Error> {
+        let cipher = cbc::Encryptor::<aes::Aes256>::new(&self.key.bytes.into(), &self.iv.into());
+        cipher
+            .encrypt_padded_b2b_mut::<Pkcs7>(plaintext, buffer)
+            .unwrap();
+        Ok(())
+    }
+
+    fn decrypt(&self, ciphertext: &[u8], buffer: &mut [u8]) -> Result<(), signature::Error> {
+        let cipher = cbc::Decryptor::<aes::Aes256>::new(&self.key.bytes.into(), &self.iv.into());
+        cipher
+            .decrypt_padded_b2b_mut::<Pkcs7>(ciphertext, buffer)
+            .unwrap();
+        Ok(())
+    }
+}
+
+///
+/// Aes128 in GCM mode (Authenticated)
+///
+pub struct Aes128Gcm<'a, NonceSize: ArrayLength<u8>> {
+    pub iv: &'a GenericArray<u8, NonceSize>,
```

Comment:
qq: how about having a similar `iv` field type (bytearray), to have consistency with the other schemes?